### ISSUE TITLE
Initialize methods can have non-void returns when sig is defined as block

### ISSUE THREAD

**Issue by jvilk-stripe** [sorbet/sorbet#5030]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20strict%0A%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20do%0A%20%20%20%20params%28a%3A%20Integer%29%0A%20%20%20%20returns%28A%29%0A%20%20end%0A%20%20def%20initialize%28a%29%0A%20%20%20%20super%0A%20%20end%0Aend%0A)

```ruby
# typed: strict

class A
  extend T::Sig

  sig do
    params(a: Integer)
    returns(A)
  end
  def initialize(a)
    super
  end
end

```

#### Observed output

No errors.

#### Expected behavior

Error that `initialize` should be `void.

- - -

Note the typo: A `.` is missing before `returns`. Inserting the `.` causes the expected error to appear.

Removing the call to `super` causes a Sorbet error because Sorbet expects the method to return `A`.

**Comment by jez** [comment]
We should just remove the non-dot format for sigs. It was added in haste to attempt to settle a debate, the debate has subsided and the `params.returns` style is clearly in more wide use.

**Comment by elliottt** [comment]
Fixed in #5055
